code/game/turfs/open/cliff.dm

```diff
@@ -0,0 +1,132 @@
+/// A cliff tile from where people can fall. Should generally fall downwards, but you can change it if you want
+/turf/open/cliff
+	icon_state = "cliff"
+	icon = 'icons/turf/cliff/cliff.dmi'
+	density = TRUE
+	/// From our perspective, where does someone need to stand to be able to fall from us? Cardinal only, we do the diagonals automatically
+	var/can_fall_from_direction = NORTH
+	/// If we fall, in which direction?
+	var/fall_direction = SOUTH
+	/// Directions to which we can move towards while grinding down
+	var/valid_move_dirs = SOUTH | WEST | EAST | SOUTHWEST | SOUTHEAST
+	/// Speed at which we fall / traverse downwards
+	var/fall_speed = 0.2 SECONDS
+	/// Movables that can move freely on cliffs
+	var/list/protected_types = list(/obj/projectile, /obj/effect, /mob/dead)
+	/// Do we draw a tile as underlay for half tiles?
+	var/turf/underlay_tile
+	/// The pixel x of the underlay image
+	var/undertile_pixel_x = 0
+	/// The pixel y of the underlay image
+	var/undertile_pixel_y = 0
+	/// if given, sets the underlays plane to this
+	var/underlay_plane
+
+/turf/open/cliff/Initialize(mapload)
+	. = ..()
+
+	RegisterSignal(src, COMSIG_TURF_MOVABLE_THROW_LANDED, PROC_REF(on_turf_movable_throw_landed))
+
+	if(underlay_tile)
+		var/image/underlay = image(icon_state = initial(underlay_tile.icon_state), icon = initial(underlay_tile.icon))
+		underlay.pixel_x = undertile_pixel_x //if there's a pixel offset, correct it because we should be lined up with the grid
+		underlay.pixel_y = undertile_pixel_y
+		SET_PLANE(underlay, underlay_plane || plane, src)
+		underlays += underlay
```

Comment:
```suggestion
	if(isnull(underlay_tile))
		return
	var/image/underlay = image(icon_state = initial(underlay_tile.icon_state), icon = initial(underlay_tile.icon))
	underlay.pixel_x = undertile_pixel_x //if there's a pixel offset, correct it because we should be lined up with the grid
	underlay.pixel_y = undertile_pixel_y
	SET_PLANE(underlay, underlay_plane || plane, src)
	underlays += underlay
```
I'm early return pilled